src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -602,6 +594,35 @@ class IntervalExpressionSegment(BaseSegment):
 )
 
 
+class ObjectReferenceSegment(ansi.ObjectReferenceSegment):
+    """A reference to an object.
+
+    https://dev.mysql.com/doc/refman/8.0/en/account-names.html
+    https://dev.mysql.com/doc/refman/8.0/en/role-names.html
+    """
+
+    match_grammar: Matchable = Sequence(
```

Comment:
This is a lot more complicated in ANSI with `DotSegments` that you've lost now. Do you need them too?

Reply:
Ah, my bad ... is this the proper thing to do? https://github.com/sqlfluff/sqlfluff/pull/3286/commits/19bafe55e9bb124216577215bb3a5a0dd8ae4f3e

Reply:
I would have redefined the whole thing. But this is much cleverer!

Reply:
Actually I don't think you did need these. DotSegments make sense for other references (tables, views...etc.) but not for users I don't think. Cleaning up in #3287 along with a few other changes.